x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -184,3 +239,21 @@ fn get_bytes(key: &[u8]) -> Option<Vec<u8>> {
         Some(ptr.into())
     }
 }
+
+trait Sealed {}
+
+#[allow(private_bounds)]
+pub trait IntoPairs: Sealed {
+    fn into_pairs(self) -> impl IntoIterator<Item = Result<(CacheKey, CacheValue), Error>>;
+}
+
+impl_to_pairs!(10, Schema);
```

Comment:
initially unclear why we need to define these, will look into macro more

Reply:
Use `cargo expand` and you will see haha
```rust
    impl<K0: Schema, K1: Schema> Sealed for ((K0, K0::Value), (K1, K1::Value)) {}
    impl<K0: Schema, K1: Schema> IntoPairs for ((K0, K0::Value), (K1, K1::Value)) {
        fn into_pairs(
            self,
        ) -> impl IntoIterator<Item = Result<(CacheKey, CacheValue), Error>> {
            [
                crate::borsh::to_vec(&self.0.1)
                    .map(|value| (Box::from(to_key(self.0.0).as_ref()), value))
                    .map_err(|_| Error::Serialization),
                crate::borsh::to_vec(&self.1.1)
                    .map(|value| (Box::from(to_key(self.1.0).as_ref()), value))
                    .map_err(|_| Error::Serialization),
            ]
        }
    }
    impl<K0: Schema> Sealed for ((K0, K0::Value),) {}
    impl<K0: Schema> IntoPairs for ((K0, K0::Value),) {
        fn into_pairs(
            self,
        ) -> impl IntoIterator<Item = Result<(CacheKey, CacheValue), Error>> {
            [
                crate::borsh::to_vec(&self.0.1)
                    .map(|value| (Box::from(to_key(self.0.0).as_ref()), value))
                    .map_err(|_| Error::Serialization),
            ]
        }
    }
}
```
That's just `impl_to_pairs(2, Schema)` and `impl_to_pairs(1, Schema)`...